.clang-tidy

```diff
@@ -1,5 +1,14 @@
 ---
-Checks:          'bugprone-throw-keyword-missing,clang-*,performance*,-performance-move-const-arg,misc-unused-parameters,-performance-avoid-endl'
+# https://clang.llvm.org/extra/clang-tidy/checks/list.html
+Checks: >
+    bugprone-throw-keyword-missing,
+    clang-*,
+    performance*,
+    -performance-move-const-arg,
+    -performance-avoid-endl,
+    misc-unused-parameters,
+    modernize-make-shared,
```

Comment:
`std::make_shared<>` makes only one allocation instead of two in `std::shared_ptr<T>(new T())`

Reply:
allocation of the pointer right, not the underlying object? `make_shared` just forwards the args to a constructor internally where as the latter passes the pointer in and copies it to its internal pointer member. have i understood that wrong? copying a pointer has to be incredibly trivial maybe even elided by some compilers.

Reply:
> copying a pointer has to be incredibly trivial maybe even elided by some compilers.

Ah I wish it could be the case... In reality it leads to atomic operation (or even mutex lock/unlock) that usually resets instruction conveyor for all CPU cores, wasting around 30 cycles for nothing...